### PR TITLE
Implement popover for cataog sources with disabled status

### PR DESCRIPTION
**Pull request by TheRealJon**
Add popover to 'Disabled' status for catalog sources on the OperatorHub config details page.

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx [hunk 1]
```diff
@@ -381,33 +381,65 @@ const CatalogSourceList: React.FC<TableProps> = (props) => (
   />
 );
 
+const DisabledPopover = ({ operatorHub, sourceName }) => {
+  const [isVisible, setIsVisible] = React.useState(null);
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx]:
```suggestion
  const [isVisible, setVisible] = React.useState(null);
```

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx [hunk 2]
```diff
@@ -381,33 +381,65 @@ const CatalogSourceList: React.FC<TableProps> = (props) => (
...
+  const close = React.useCallback(() => {
+    setIsVisible(false);
+  }, [setIsVisible]);
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx]: I don't think we need this in the dependencies

File: frontend/packages/console-shared/src/components/status/PopoverStatus.tsx [hunk 3]
```diff
@@ -8,6 +9,8 @@ const PopoverStatus: React.FC<PopoverStatusProps> = ({
   icon,
   activeIcon,
   children,
+  isVisible = null,
+  shouldClose = null,
```
Comment (spadgett) [inline frontend/packages/console-shared/src/components/status/PopoverStatus.tsx]: Do we need to default to null? I'd think `undefined` would be OK

**Comment by spadgett** [comment]
/override ci/prow/e2e-gcp

Test has been removed

File: frontend/packages/console-shared/src/components/status/PopoverStatus.tsx (hunk 3 above)
Comment (TheRealJon) [reply frontend/packages/console-shared/src/components/status/PopoverStatus.tsx]: There is a conditional check in the underlying component which will ignore the isVisible prop as long as it `=== null`. I have a feeling that letting it default to undefined might cause some undesired behavior.

Comment (spadgett) [reply frontend/packages/console-shared/src/components/status/PopoverStatus.tsx]: That almost sounds like a bug in the PF component. It treats `null` and `undefined` differently?

**Pull request merged**